src/System.Data.SqlClient/src/System/Data/Locale/LocaleMapper.Unix.cs

```diff
@@ -450,7 +451,13 @@ internal class LocaleMapper
 
         public static string LcidToLocaleNameInternal(int lcid)
         {
-            return s_mapper[lcid].LocaleName;
+            LocaleCodePage value;
+            if (s_mapper.TryGetValue(lcid, out value))
+            {
+                return value.LocaleName;
+            }
+
+            throw new CultureNotFoundException(nameof(lcid), lcid.ToString(), message: null);
```

Comment:
These changes are to make the Unix implementation behave the same as Windows. Other methods below may need some additional work, but I'll look into that for a separate PR as they aren't related to the changes in this PR.